src/EditorFeatures/TestUtilities/PullMemberUp/TestPullMemberUpService.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp.Dialog;
+
+namespace Microsoft.CodeAnalysis.Test.Utilities.PullMemberUp
+{
+    internal class TestPullMemberUpService : IPullMemberUpOptionsService
+    {
+        private IEnumerable<(string member, bool makeAbstract)> SelectedMembers { get; }
+        
+        private string TargetBaseTypeName { get; }
+
+        public TestPullMemberUpService(IEnumerable<(string member, bool makeAbstract)> selectedMembers, string targetBaseTypeName)
+        {
+            SelectedMembers = selectedMembers;
+            TargetBaseTypeName = targetBaseTypeName;
+        }
+
+        public PullMemberDialogResult GetPullTargetAndMembers(SemanticModel semanticModel, ISymbol selectedNodeSymbol)
+        {
+            IEnumerable<(ISymbol member, bool makeAbstract)> selectedMember = default;
+            var members = selectedNodeSymbol.ContainingType.GetMembers().Where(
+                    member => {
+                        if (member is IMethodSymbol methodSymbol)
+                        {
+                            return methodSymbol.MethodKind == MethodKind.Ordinary;
+                        }
+                        else if (member is IFieldSymbol fieldSymbol)
+                        {
+                            return !member.IsImplicitlyDeclared;
+                        }
+                        else
+                        {
+                            return member.Kind == SymbolKind.Property || member.Kind == SymbolKind.Event;
```

Comment:
i've seen this code in multiple locations.  We should not be duplicating htis all over the place (especially in tests).  This code should call into the exact same helper that determines the data that populate the UI.

Specifically: the UI should *sit* on top of code that is entirely non-ui related which computes this stuff.  You can then have this "test UI" sit on the exact same common code.  The "test ui" then just pretends it's the xaml UI, but otherwise calls into the exact logic shared by the normal UI.  The normal UI should do almost *nothing* semantically interesting since otherwise that will be super hard to test.